Some projects need to add <InterceptorsNamespaces> with the latest aspnet builds

Example in Orchard:

```
/tmp/benchmarks-agent/benchmarks-server-1/iaulnbd3.afr/src/OrchardCore.Modules/OrchardCore.OpenId/obj/Release/net10.0/Microsoft.AspNetCore.Http.ValidationsGenerator/Microsoft.AspNetCore.Http.ValidationsGenerator.ValidationsGenerator/ValidatableInfoResolver.g.cs(79,10): error CS9137: The 'interceptors' feature is not enabled in this namespace. Add '<InterceptorsNamespaces>$(InterceptorsNamespaces);Microsoft.AspNetCore.Http.Validation.Generated</InterceptorsNamespaces>' to your project. [/tmp/benchmarks-agent/benchmarks-server-1/iaulnbd3.afr/src/OrchardCore.Modules/OrchardCore.OpenId/OrchardCore.OpenId.csproj::TargetFramework=net10.0]
```

https://github.com/OrchardCMS/OrchardCore/pull/17685/files

--- a/src/WebSdk/ProjectSystem/Targets/Microsoft.NET.Sdk.Web.ProjectSystem.targets
+++ b/src/WebSdk/ProjectSystem/Targets/Microsoft.NET.Sdk.Web.ProjectSystem.targets
@@ -65,11 +65,11 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <PropertyGroup>
     <!-- Set the namespaces emitted by the RequestDelegateGenerator for interception when applicable. -->
-    <InterceptorsNamespaces Condition="'$(EnableRequestDelegateGenerator)' == 'true'">$(InterceptorsNamespaces);Microsoft.AspNetCore.Http.Generated</InterceptorsNamespaces>
+    <InterceptorsPreviewNamespaces Condition="'$(EnableRequestDelegateGenerator)' == 'true'">$(InterceptorsPreviewNamespaces);Microsoft.AspNetCore.Http.Generated</InterceptorsPreviewNamespaces>
     <!-- Set the namespaces emitted by the ConfigurationBindingGenerator for interception when applicable. -->
-    <InterceptorsNamespaces Condition="'$(EnableConfigurationBindingGenerator)' == 'true'">$(InterceptorsNamespaces);Microsoft.Extensions.Configuration.Binder.SourceGeneration</InterceptorsNamespaces>
+    <InterceptorsPreviewNamespaces Condition="'$(EnableConfigurationBindingGenerator)' == 'true'">$(InterceptorsPreviewNamespaces);Microsoft.Extensions.Configuration.Binder.SourceGeneration</InterceptorsPreviewNamespaces>
     <!-- Set the namespaces emitted by the ValidationsGenerator for interception when applicable (.NET 10 and above). -->
-    <InterceptorsNamespaces Condition="'$(_TargetFrameworkVersionWithoutV)' != '' and '$(_TargetFrameworkVersionWithoutV)' >= '10.0'">$(InterceptorsNamespaces);Microsoft.AspNetCore.Http.Validation.Generated</InterceptorsNamespaces>
+    <InterceptorsPreviewNamespaces Condition="'$(_TargetFrameworkVersionWithoutV)' != '' and '$(_TargetFrameworkVersionWithoutV)' >= '10.0'">$(InterceptorsPreviewNamespaces);Microsoft.AspNetCore.Http.Validation.Generated</InterceptorsPreviewNamespaces>
   </PropertyGroup>
 
   <!--

--- a/src/WebSdk/ProjectSystem/Targets/Microsoft.NET.Sdk.Web.ProjectSystem.targets
+++ b/src/WebSdk/ProjectSystem/Targets/Microsoft.NET.Sdk.Web.ProjectSystem.targets
@@ -69,7 +69,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     <!-- Set the namespaces emitted by the ConfigurationBindingGenerator for interception when applicable. -->
     <InterceptorsPreviewNamespaces Condition="'$(EnableConfigurationBindingGenerator)' == 'true'">$(InterceptorsPreviewNamespaces);Microsoft.Extensions.Configuration.Binder.SourceGeneration</InterceptorsPreviewNamespaces>
     <!-- Set the namespaces emitted by the ValidationsGenerator for interception when applicable (.NET 10 and above). -->
-    <InterceptorsPreviewNamespaces Condition="'$(_TargetFrameworkVersionWithoutV)' != '' and '$(_TargetFrameworkVersionWithoutV)' >= '10.0'">$(InterceptorsPreviewNamespaces);Microsoft.AspNetCore.Http.Validation.Generated</InterceptorsPreviewNamespaces>
+    <InterceptorsPreviewNamespaces Condition="'$(_TargetFrameworkVersionWithoutV)' != '' and $([MSBuild]::VersionGreaterThanOrEquals('$(_TargetFrameworkVersionWithoutV)', '10.0'))">$(InterceptorsPreviewNamespaces);Microsoft.AspNetCore.Http.Validation.Generated</InterceptorsPreviewNamespaces>
   </PropertyGroup>
 
   <!--

--- a/src/WebSdk/ProjectSystem/Targets/Microsoft.NET.Sdk.Web.ProjectSystem.targets
+++ b/src/WebSdk/ProjectSystem/Targets/Microsoft.NET.Sdk.Web.ProjectSystem.targets
@@ -65,9 +65,11 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <PropertyGroup>
     <!-- Set the namespaces emitted by the RequestDelegateGenerator for interception when applicable. -->
-    <InterceptorsPreviewNamespaces Condition="'$(EnableRequestDelegateGenerator)' == 'true'">$(InterceptorsPreviewNamespaces);Microsoft.AspNetCore.Http.Generated</InterceptorsPreviewNamespaces>
+    <InterceptorsNamespaces Condition="'$(EnableRequestDelegateGenerator)' == 'true'">$(InterceptorsNamespaces);Microsoft.AspNetCore.Http.Generated</InterceptorsNamespaces>
     <!-- Set the namespaces emitted by the ConfigurationBindingGenerator for interception when applicable. -->
-    <InterceptorsPreviewNamespaces Condition="'$(EnableConfigurationBindingGenerator)' == 'true'">$(InterceptorsPreviewNamespaces);Microsoft.Extensions.Configuration.Binder.SourceGeneration</InterceptorsPreviewNamespaces>
+    <InterceptorsNamespaces Condition="'$(EnableConfigurationBindingGenerator)' == 'true'">$(InterceptorsNamespaces);Microsoft.Extensions.Configuration.Binder.SourceGeneration</InterceptorsNamespaces>
+    <!-- Set the namespaces emitted by the ValidationsGenerator for interception when applicable (.NET 10 and above). -->
+    <InterceptorsNamespaces Condition="'$(_TargetFrameworkVersionWithoutV)' != '' and '$(_TargetFrameworkVersionWithoutV)' >= '10.0'">$(InterceptorsNamespaces);Microsoft.AspNetCore.Http.Validation.Generated</InterceptorsNamespaces>
   </PropertyGroup>
 
   <!--

--- a/test/Microsoft.NET.Build.Tests/GivenThatWeWantToUseAnalyzers.cs
+++ b/test/Microsoft.NET.Build.Tests/GivenThatWeWantToUseAnalyzers.cs
@@ -73,6 +73,21 @@ namespace Microsoft.NET.Build.Tests
             VerifyInterceptorsFeatureProperties(asset, expectEnabled: true, "Microsoft.AspNetCore.Http.Generated", "Microsoft.Extensions.Configuration.Binder.SourceGeneration");
         }
 
+        [Theory]
+        [InlineData("net10.0", true)]
+        [InlineData("net9.0", false)]
+        [InlineData("net8.0", false)]
+        public void It_enables_validationsgenerator_correctly_for_TargetFramework(string targetFramework, bool expectEnabled)
+        {
+            var asset = _testAssetsManager
+                .CopyTestAsset("WebApp")
+                .WithSource()
+                .WithTargetFramework(targetFramework);
+
+            VerifyValidationsGeneratorIsUsed(asset, expectEnabled);
+            VerifyInterceptorsFeatureProperties(asset, expectEnabled, "Microsoft.AspNetCore.Http.Validation.Generated");
+        }
+
         [Fact]
         public void It_enables_requestdelegategenerator_and_configbindinggenerator_for_PublishTrimmed()
         {
@@ -115,13 +130,16 @@ namespace Microsoft.NET.Build.Tests
         private void VerifyConfigBindingGeneratorIsUsed(TestAsset asset, bool? expectEnabled)
             => VerifyGeneratorIsUsed(asset, expectEnabled, "Microsoft.Extensions.Configuration.Binder.SourceGeneration.dll");
 
+        private void VerifyValidationsGeneratorIsUsed(TestAsset asset, bool? expectEnabled)
+            => VerifyGeneratorIsUsed(asset, expectEnabled, "Microsoft.AspNetCore.Http.ValidationsGenerator.dll");
+
         private void VerifyInterceptorsFeatureProperties(TestAsset asset, bool? expectEnabled, params string[] expectedNamespaces)
         {
             var command = new GetValuesCommand(
                 Log,
                 asset.Path,
                 ToolsetInfo.CurrentTargetFramework,
-                "InterceptorsPreviewNamespaces",
+                "InterceptorsNamespaces",
                 GetValuesCommand.ValueType.Property);
 
             command

--- a/test/Microsoft.NET.Build.Tests/GivenThatWeWantToUseAnalyzers.cs
+++ b/test/Microsoft.NET.Build.Tests/GivenThatWeWantToUseAnalyzers.cs
@@ -139,7 +139,7 @@ namespace Microsoft.NET.Build.Tests
                 Log,
                 asset.Path,
                 ToolsetInfo.CurrentTargetFramework,
-                "InterceptorsNamespaces",
+                "InterceptorsPreviewNamespaces",
                 GetValuesCommand.ValueType.Property);
 
             command

